### PR TITLE
[WIP] docs: Creating Folder Structure inside readme 

### PR DESCRIPTION
**Pull request by himanshu010**
fixes: #1297 
**What kind of change does this PR introduce?**

docs related change

**Did you add tests for your changes?**
N/A

**If relevant, did you update the documentation?**
N/A

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -107,6 +107,224 @@ You can read more about [Scaffolding](https://webpack.js.org/guides/scaffolding)
 
 The webpack family welcomes any contributor, small or big. We are happy to elaborate, guide you through the source code and find issues you might want to work on! To get started have a look at our [documentation on contributing](./.github/CONTRIBUTING.md).
 
+### Folder Structure
+```
+├───.git
```
Comment (snitin315) [inline README.md]: You can remove .git

**Comment by snitin315** [comment]
Also use `yarn commit` for proper commit message. You can find more info in the `CONTRIBUTING.md` file.

**Comment by himanshu010** [comment]
> Also use `yarn commit` for proper commit message. You can find more info in the `CONTRIBUTING.md` file.

It is comitted directly from GitHub. I will take care of it from next time.

File: README.md (hunk 1 above)
Comment (himanshu010) [reply README.md]: sure

**Comment by anshumanv** [comment]
> It is committed directly from GitHub. I will definitely take care of it next time.

Please avoid that, commit guidelines are in place for a reason.

**Comment by himanshu010** [comment]
> > It is committed directly from GitHub. I will definitely take care of it next time.
> 
> Please avoid that, commit guidelines are in place for a reason.



> > It is committed directly from GitHub. I will definitely take care of it next time.
> 
> Please avoid that, commit guidelines are in place for a reason.

Do I have to change it now?

**Comment by himanshu010** [comment]
> @himanshu010 how about creating a new file for folder structure?

Yeah, I think that would be great. We can later hyperlink the file in the readme.

**Comment by himanshu010** [comment]
@ematipico Yeah folder description is a must in a folder structure. I made this PR so that the reviewing team can understand how the folder structure is going to look like(Basically this is an example) as I want some reviews from them. I am working on the description right now and I will make a pr as soon as possible.

>@himanshu010 how about creating a new file for folder structure?

I think that this would be nicer than placing it in the readme. We can hyperlink it there like this.
Screenshot (59)
what are your views on this @ematipico

**Comment by himanshu010** [comment]
> Also, there are folders that should not be there: for instances `lib` inside the packages written in TypeScript. These are folders that are created only when publishing the package (and for integration testing).
> 
> I also think that the tests folder should stop there. `__tests__ (contains unit test of the source code)` and that's it. I believe it's obvious.

Yes, I am totally convinced by your words but according to the definitions, a 'folder structure' must have to contain all the folders inside it.

**Comment by ematipico** [comment]
This is a tedious work though. If someone changes the name of a file of a directory, this folder structure won't be good anymore. Do you see my point?

If you really want go through this PR, I'd suggest evaluating how to change this file dynamically. (that you can do with a different PR). Having a script that creates the tree file and dynamically update the file.

**Comment by himanshu010** [comment]
> If you really want go through this PR, I'd suggest evaluating how to change this file dynamically. (that you can do with a different PR). Having a script that creates the tree file and dynamically update the file. You will have to create a new issue and doing a new PR where we are going to make this dynamic. Do you want to continue? It's going to be a nice work!

Yes, I definitely want to continue.

**Comment by himanshu010** [comment]
>If you really want go through this PR, I'd suggest evaluating how to change this file dynamically. (that you can do with a different PR)

I will create two pull requests associated with it.

**Pull request closed without merging**